base/http_listener.go

```diff
@@ -27,6 +34,25 @@ func ListenAndServeHTTP(addr string, connLimit int, certFile *string, keyFile *s
 		if err != nil {
 			return err
 		}
+
+		if clientCerts != nil {
+			// TLS client certificate support:
+			clientCAData, err := ioutil.ReadFile(clientCerts.CA_cert)
+			if err != nil {
+				return err
+			}
+			config.ClientCAs = x509.NewCertPool()
+			if !config.ClientCAs.AppendCertsFromPEM(clientCAData) {
+				return ErrInvalidCertificate
```

Comment:
If AppendCertsFromPEM hits an error during ParseCertificate, it doesn't return any details about that failure.  In the event AppendCertsFromPEM returns false, could we unmarshal clientCAData into pem.Block and invoke ParseCertificate to bubble up the failure information?